### ISSUE TITLE
Store pre-trained models as workflow artifacts

### ISSUE THREAD

**Issue by ssinad** [ivadomed/ivadomed#801]
### Description of the feature

The [pre-trained models](https://github.com/ivadomed/ivadomed/blob/master/docs/source/pretrained_models.rst) in the following repositories could be stored as [workflow artifacts](https://docs.github.com/en/actions/guides/storing-workflow-data-as-artifacts).

### Alternatives

There are definitely alternatives for storing the models such as [Git LFS](https://git-lfs.github.com).
Suggestions are welcome!

**Comment by jcohenadad** [comment]
@ssinad good thoughts-- we have been discussing that for the past few years-- there are pros/cons-- some of the discussions can be found [there](https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues?q=is%3Aissue+git-lfs+).

**Comment by ssinad** [comment]
I also found [DVC](https://dvc.org/doc/use-cases/versioning-data-and-model-files).

**Comment by ssinad** [comment]
I've also heard of [MLflow](https://mlflow.org), but I am not sure if it addresses this specific issue or not.

**Comment by jcohenadad** [comment]
> I also found [DVC](https://dvc.org/doc/use-cases/versioning-data-and-model-files).

DVC looks very interesting. I would be very curious to hear @kousu's thoughts on this technology

**Comment by kousu** [comment]
The biggest problem with Artifacts is that they are meant to be temporary. They're meant for passing data from one script to another -- which usually runs on a separate CI server -- not for long term storage. For long term storage you need to upload to a repo like one of [github's package servers](https://docs.github.com/en/packages), or a bigger package server like maven, npm, dockerhub, pypi, or 

In SCT I've been working (in fits and spurts) towards using pypi for us, because I figure the rest of the project is in python so sticking with the python ecosystem will simplify everything: https://github.com/neuropoly/spinalcordtoolbox/issues/2669. I'm working towards packaging everything -- template images, SVM models, neural network models -- as python packages. We had some more thoughts [just last week](https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3396) about handling neural networks specifically; particularly, I think they can fit into the standard build/package/release cycle of other kinds of software, they don't need to be such a niche thing; @dyt811 has a spreadsheet of nearly 50 neural network pipeline systems, but none of them stands out to me because they all want to solve your problem by making you do everything on their servers.

> DVC looks very interesting. I would be very curious to hear @kousu's thoughts on this technology

I've stumbled across DVC before but I've never taken the time to look into it. It's cool it's built on git; seems like it's a direct competitor to git-annex maybe? Maybe better done?

My thoughts on this, currently, are still [it's worth taking the time to see how far git itself has evolved](https://github.com/neuropoly/data-management/issues/68); I think we can probably get away with just using git, with suitable configs, instead of pulling in any extra software or workflows. But maybe DVC has thought through all these issues?

**Comment by dyt811** [comment]
For those that are curious, this is what i do during spare time looking at [ML/DL experiment management](https://docs.google.com/spreadsheets/d/1rOaa-x1cRveqg-1fK0BJkPUxKl736DUQA8I6fYeTvWg/edit?usp=drivesdk).

Yes. MLflow and DVC are both on there. Last updated this April.

TLDR: most of these are experiment tracking, some support artifact storing. Many can deploy on your own platforms. I usually recommend trying proof of concept before commiting to any specific ones.
